source/BrightcoveOS .NET-MAPI-Wrapper/Model/Items/BrightcoveCaptionSource.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web.Script.Serialization;
+using BrightcoveMapiWrapper.Serialization;
+using BrightcoveMapiWrapper.Util;
+
+namespace BrightcoveMapiWrapper.Model.Items
+{
+    /// <summary>
+    /// A source that provides captions for a video
+    /// </summary>
+    /// <remarks>
+    /// See https://docs.brightcove.com/en/video-cloud/media/references/reference.html#CaptionSource
+    /// </remarks>
+    public class BrightcoveCaptionSource : BrightcoveItem, IJavaScriptConvertable
+    {
+        /// <summary>
+        /// A Boolean indicating whether a CaptionSource is usable.
+        /// </summary>
+        public bool Complete { get; set; }
+
+        /// <summary>
+        /// The name of the caption source, which will be displayed in the Media module.
+        /// </summary>
+        public string DisplayName { get; set; }
+
+        /// <summary>
+        /// A number that uniquely identifies this CaptionSource object, 
+        /// assigned by Video Cloud when this object is created.
+        /// </summary>
+        public long? Id { get; set; }
+
+        /// <summary>
+        /// A Boolean indicating whether or not this CaptionSource is hosted on a remote server, 
+        /// as opposed to hosted by Brightcove.
+        /// </summary>
+        public bool IsRemote { get; set; }
+
+        /// <summary>
+        /// The complete path to the file.
+        /// </summary>
+        public string Url { get; set; }
+
+        public IDictionary<string, object> Serialize(JavaScriptSerializer serializer)
+        {
+            IDictionary<string, object> serialized = new Dictionary<string, object>();
+
+            serialized["complete"] = Complete;
+            serialized["displayName"] = DisplayName;
+            serialized["id"] = Id ?? 0;
```

Comment:
Same concern about read-only fields. No need to set them, assuming Searilize is used to send data.